mesonbuild/modules/gnome.py

```diff
@@ -22,22 +22,34 @@
 from .. import dependencies
 from .. import mlog
 from .. import mesonlib
+from .. import interpreter
 
+glib_version = None
 girwarning_printed = False
 gresource_warning_printed = False
 
 class GnomeModule:
 
+    def get_glib_version(self, state):
+        global glib_version
+        if glib_version is None:
+            glib_dep = dependencies.PkgConfigDependency('glib-2.0', state.environment, {})
```

Comment:
This checks the version of the build machine, which will fail in cross compilation if the host machine has different version of glib. It should probably get the pkg-config binary name from cross info or just use `pkg-config` if not cross compiling and just directly call `modversion`.